### PR TITLE
chore: drop support for node 10 and use latest node 16 features

### PR DESCRIPTION
**Pull request by ImRodry**
This PR updates CI to run on node versions 12, 14 and 16 (all the ones supported by node) and refactors some code to use the latest features from es2021. This doesn't need a release, it's a simple QOL update

### CODE REVIEW

**Comment by ImRodry** [comment]
node v12 will reach its end-of-life on April 30th (https://nodejs.org/en/about/releases/) so when that comes around we can also deprecate node 12 and update to npm v7 and package-lock v2. I'll leave it up to you to decide if you wanna wait until then or merge this now and then make that switch later on

**Review by yevheniyJ — APPROVED** [review]

**Pull request merged**
